cmd/api/logging/logging.go

```diff
@@ -0,0 +1,178 @@
+func LogRetrieval() gin.HandlerFunc {
+	// FIXME For now the queries to the logging backend server are done insecurely. Needed for the current test env.
+	// #nosec G402
+	client := http.Client{
+		Transport: otelhttp.NewTransport(&http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true}})}
+	return func(c *gin.Context) {
+		user := c.GetString(userKey)
+		password := c.GetString(passwordKey)
+		if user == "" || password == "" {
+			loggingCredsErr := fmt.Errorf(
+				"unexpected error. Logging credentials are unset")
+			abort.Abort(c, loggingCredsErr, http.StatusInternalServerError)
+			return
+		}
+		logUrl := c.GetString("logURL")
+		jsonPath := c.GetString("jsonPath")
+
+		if logUrl == "" {
+			abort.Abort(c, fmt.Errorf("no log URL found"), http.StatusNotFound)
+			return
+		}
+		slog.Info("Retrieving logs", "logURL", logUrl)
+
+		var jsonPathParser jp.Expr
+		var errJsonPath error
+
+		if jsonPath != "" {
+			jsonPathParser, errJsonPath = jp.ParseString(jsonPath)
+			if errJsonPath != nil {
+				abort.Abort(c, fmt.Errorf("invalid jsonPath %s for url %s", jsonPath, logUrl),
+					http.StatusInternalServerError)
+				return
+			}
+		}
+
+		request, errReq := http.NewRequest(http.MethodGet, logUrl, nil)
+		if errReq != nil {
+			abort.Abort(c, errReq, http.StatusInternalServerError)
+			return
+		}
+
+		request.SetBasicAuth(user, password)
+
+		response, errReq := client.Do(request)
+		if errReq != nil {
+			abort.Abort(c, errReq, http.StatusInternalServerError)
+			return
+		}
+
+		defer response.Body.Close()
+
+		reader := bufio.NewReader(response.Body)
+		line, readErr := readLine(reader)
+		var logsReturned bool
+		for readErr == nil {
+			parsedLine, errParseLine := parseLine(line, jsonPathParser)
+			if errParseLine != nil {
+				abort.Abort(c, errParseLine, http.StatusInternalServerError)
+				return
+			}
+			if parsedLine != "" {
+				logsReturned = true
+				c.String(http.StatusOK, parsedLine+"\n")
+			}
+			line, readErr = readLine(reader)
+		}
+
+		if !logsReturned {
+			abort.Abort(c, fmt.Errorf("no parsed logs for the json path: %s", jsonPathParser.String()),
+				http.StatusNotFound)
+		}
+	}
+}
+
+func parseLine(line []byte, jsonPathParser jp.Expr) (string, error) {
+	if jsonPathParser == nil {
+		return string(line), nil
+	}
+
+	var jsonLine any
+	var errJson error
+	var result string
+
+	jsonLine, errJson = oj.Parse(line)
+	if errJson != nil {
+		return result, errJson
+	}
+	for _, res := range jsonPathParser.Get(jsonLine) {
+		if result == "" {
+			result = res.(string)
+		} else {
+			result = fmt.Sprintf("%s\n%s", result, res.(string))
+		}
+	}
+	return result, nil
+}
+
+func readLine(reader *bufio.Reader) ([]byte, error) {
+	var (
+		isPrefix            = true
+		err           error = nil
+		tmpLine, line []byte
+	)
+	for isPrefix && err == nil {
+		tmpLine, isPrefix, err = reader.ReadLine()
+		line = append(line, tmpLine...)
+	}
+	return line, err
+}
```

Comment:
Any reason not to simply use `bufio.ReadBytes('\n')` rather than writing your own?

Reply:
I made the decission based on [this stackoverflow answer](https://stackoverflow.com/questions/8757389/reading-a-file-line-by-line-in-go#answer-8758113). The points I considered were:
* If the last log line doesn't end with a `\n` won't be parsered by `bufio.ReadBytes('\n')` or `bufio.ReadString('\n')`
* `bufio.ReadLine` doesn't return the full line when it's too large that's why I need to do some extra process to the returned results of `reader.ReadLine()`

Reply:
hmm, right... I'm trying to change it now (I found some issues on the related unit test so I'm trying to address both)

Reply:
Fixed